crates/curp/src/server/storage/wal/codec.rs

```diff
@@ -282,17 +285,18 @@ where
                     .unwrap_or_else(|_| unreachable!("serialization should never fail"));
                 let len = entry_bytes.len();
                 assert_eq!(len >> 56, 0, "log entry length: {len} too large");
-                let len_bytes = len.to_le_bytes().into_iter().take(7);
-                let header = std::iter::once(self.frame_type()).chain(len_bytes);
-                header.chain(entry_bytes).collect()
+                let mut bytes = Vec::with_capacity(1 + 7 + entry_bytes.len());
```

Comment:
Can we replace `1` with `std::mem::size_of_val(&self.frame_type())`?  And using a constant to replace the magic number `7`.